src/views/forms/CancelAllOrdersInMarketForm.tsx

```diff
@@ -0,0 +1,198 @@
+  | { type: 'error'; errorKey?: string };
+
+export const CancelAllOrdersInMarketForm = ({
+  marketId,
+  onCancelComplete,
+}: CancelAllOrdersInMarketFormProps) => {
+  const stringGetter = useStringGetter();
+  const pendingPositions = useSelector(getNonZeroPendingPositions, shallowEqual);
+  const thisPendingPosition = useMemo(
+    () => pendingPositions?.find((f) => f.marketId === marketId),
+    [marketId, pendingPositions]
+  );
+  const allPending = useSelector(getPendingIsolatedOrders, shallowEqual);
+  const pendingPositionOrders = allPending[marketId] ?? EMPTY_ARR;
+  const assetsData = useSelector(getAssets, shallowEqual);
+
+  const [cancellingStatus, setCancellingStatus] = useState<Record<string, OrderCancelStatus>>({});
+  const isCancelling = useMemo(
+    () => Object.values(cancellingStatus).some((s) => s.type === 'pending'),
+    [cancellingStatus]
+  );
+  const { cancelOrder } = useSubaccount();
+  const { freeCollateral: crossFreeCollateral } = useSelector(getSubaccount, shallowEqual) ?? {};
+
+  const onCancel = useCallback(() => {
+    if (isCancelling) {
+      return;
+    }
+    setCancellingStatus(
+      zipObject(
+        pendingPositionOrders.map((p) => p.id),
+        pendingPositionOrders.map(() => ({ type: 'pending' }))
+      )
+    );
+    pendingPositionOrders.forEach((p) =>
+      cancelOrder({
+        orderId: p.id,
+        onSuccess: () => {
+          setCancellingStatus((old) => ({ ...old, [p.id]: { type: 'success' } }));
+        },
+        onError: (errorInfo) => {
+          setCancellingStatus((old) => ({
+            ...old,
+            [p.id]: { type: 'error', errorKey: errorInfo?.errorStringKey ?? undefined },
+          }));
+        },
+      })
+    );
+  }, [cancelOrder, isCancelling, pendingPositionOrders]);
+
+  useEffect(() => {
+    const allResults = Object.values(cancellingStatus);
+    if (allResults.length === 0) {
+      return;
+    }
+    // if there are errors, user should be able to see the error notifications so we won't display inline.
+    if (allResults.every((r) => r.type === 'success' || r.type === 'error')) {
+      onCancelComplete();
+    }
+  }, [cancellingStatus, onCancelComplete]);
+
+  const detailItems = useMemo(() => {
+    return [
+      {
+        key: 'open-orders',
+        label: <span>{stringGetter({ key: STRING_KEYS.OPEN_ORDERS })}</span>,
+        value: (
+          <$DiffOutput
+            type={OutputType.Number}
+            value={pendingPositionOrders.length}
+            newValue={0}
+            sign={NumberSign.Negative}
+            withDiff
+          />
+        ),
+      },
+      {
+        key: 'position-margin',
+        label: <span>{stringGetter({ key: STRING_KEYS.POSITION_MARGIN })}</span>,
+        value: (
+          <$DiffOutput
+            type={OutputType.Fiat}
+            value={thisPendingPosition?.freeCollateral?.current ?? 0}
+            newValue={0}
+            sign={NumberSign.Negative}
+            withDiff
```

Comment:
Add conditional to this boolean, we shouldn't display diff if value and newValue are the same i.e. `0 -> 0`. I know this shouldn't happen due to the nature of the component but we should guard against it.

Reply:
`value` can also be null or undefined which results in `-`. We prefer `-` to `0` in most cases.